src/analyses/README.md

```diff
@@ -121,6 +121,144 @@ is covered in a separate subsection.
 
 A basic description for how a natural loop works is here: https://web.cs.wpi.edu/~kal/PLT/PLT8.6.4.html
 
+### Reaching definitions
+
+This module implements the reaching definitions data-flow analysis
+(see https://en.wikipedia.org/wiki/Reaching_definition for basic introduction)
+on multi-threaded GOTO programs.
+
+The domain element is defined in a class `rd_range_domaint` and the actual
+analysis is implemented in a class `reaching_definitions_analysist`. Beside
+these classes there are more data structures necessary in the computation.
+We discuss all of them in the following sub-sections.   
+
+#### Struct `reaching_definitiont`
+
+Identifies a GOTO instruction where a given variable is defined (i.e. it is set
+a certain value). In consists of these data:
+- `identifier` The name of the variable which was defined.
+- `definition_at` The iterator to the GOTO instruction where the variable is
+- `bit_begin` and `bit_end` These two integers define a range of bits (i.e. the
+  begin and end bit indices) which represent the value of the variable. Clearly,
+  `0 <= bit_begin < bit_end`. However, `bit_end` can also be set a special value
+  `-`, which means infinite/unknown index.
+
+There is defined an order of instances of this structure. But its purpose is
+only for to be able to use instances as keys in ordered containers, like
+`std::map`.
```

Comment:
I wonder if this paragraph might be better phrased as something like this?

In order to use instances of this structure as keys in ordered containers, such as `std::map`, an ordering is defined.